src/panoptes/pocs/filterwheel/filterwheel.py

```diff
@@ -196,28 +213,53 @@ def move_to(self, position, blocking=False):
             self.logger.error(msg)
             raise error.PanError(msg)
 
-        position = self._parse_position(position)
-        self.logger.info("Moving {} to position {} ({})".format(
-            self, position, self.filter_names[position - 1]))
+        # Will raise a ValueError at this point if new_position is not a valid position
+        new_position = self._parse_position(new_position)
 
-        if position == self.position:
+        if new_position == self.position:
             # Already at requested position, don't go nowhere.
+            self.logger.debug(f"{self} already at position {new_position}" + \
+                              f" ({self.filter_name(new_position)})")
             return self._move_event
 
+        if new_position == self._dark_position:
+            # Moving from light into darkness... Store current position so we can revert
+            # back to it if requested with move_to_light_position()
+            self._last_light_position = self.position
+
+        self.logger.info("Moving {} to position {} ({})".format(
+            self, new_position, self.filter_name(new_position)))
         self._move_event.clear()
-        self._move_to(position)  # Private method to actually perform the move.
+        self._move_to(new_position)  # Private method to actually perform the move.
 
         if blocking:
             self._move_event.wait()
 
         return self._move_event
 
+    def move_to_dark_position(self, blocking=False):
+        """ Move to filterwheel position for taking darks. """
+        try:
+            return self.move_to(self._dark_position, blocking=blocking)
+        except ValueError:
+            msg = f"Request to move to dark position but {self} has no dark_position set."
+            raise error.NotFound(msg)
+
+    def move_to_light_position(self, blocking=False):
+        """ Return to last filterwheel position from before taking darks. """
+        try:
+            return self.move_to(self._last_light_position, blocking=blocking)
```

Comment:
```suggestion
            self.logger.info(f'Moving filter to light position after dark')
            return self.move_to(self._last_light_position, blocking=blocking)
```

Reply:
As above, this gets called regardless of whether the filterwheel is already at a light position.